easybuild/tools/toolchain/compiler.py

```diff
@@ -146,8 +146,11 @@ def set_options(self, options):
 
     def set_variables(self):
         """Set the variables"""
+        if self.arch is None:
+            self.arch = systemtools.get_cpu_architecture()
```

Comment:
hmm, aren't there any other locations where we rely on the `.arch` attribute of a `Toolchain` instance (potentially) that we should worry about?

playing the devil's advocate here, since this approach makes more sense anyway (defining `self.arch` with the result of `get_cpu_family` was already an indication that something wasn't 100% right...)